security-framework/src/os/macos/code_signing.rs

```diff
@@ -0,0 +1,269 @@
+//! Code signing services.
+
+use std::{mem::MaybeUninit, str::FromStr};
+
+use core_foundation::{
+    base::{TCFType, TCFTypeRef},
+    data::CFDataRef,
+    dictionary::CFMutableDictionary,
+    number::CFNumber,
+    string::CFString,
+    url::CFURL,
+};
+use libc::pid_t;
+use security_framework_sys::code_signing::{
+    kSecCSCheckTrustedAnchors, kSecCSConsiderExpiration, kSecCSEnforceRevocationChecks,
+    kSecCSNoNetworkAccess, kSecCSQuickCheck, kSecCSReportProgress, kSecGuestAttributeAudit,
+    kSecGuestAttributePid, SecCodeCheckValidity, SecCodeCopyGuestWithAttributes, SecCodeCopyPath,
+    SecCodeCopySelf, SecCodeGetTypeID, SecCodeRef, SecRequirementCreateWithString,
+    SecRequirementGetTypeID, SecRequirementRef, SecStaticCodeCheckValidity,
+    SecStaticCodeCreateWithPath, SecStaticCodeGetTypeID, SecStaticCodeRef,
+};
+
+use crate::{cvt, Result};
+
+bitflags::bitflags! {
+    /// Values that can be used in the flags parameter to most code signing
+    /// functions.
+    pub struct Flags: u32 {
+        const NONE = 0;
+
+        /// Consider expired certificates invalid.
+        const CONSIDER_EXPIRATION = kSecCSConsiderExpiration;
+        const ENFORCE_REVOCATION_CHECKS = kSecCSEnforceRevocationChecks;
+        const NO_NETWORK_ACCESS = kSecCSNoNetworkAccess;
+        const REPORT_PROGRESS = kSecCSReportProgress;
+        const CHECK_TRUSTED_ANCHORS = kSecCSCheckTrustedAnchors;
+        const QUICK_CHECK = kSecCSQuickCheck;
+    }
+}
+
+impl Default for Flags {
+    #[inline(always)]
+    fn default() -> Self {
+        Self::NONE
+    }
+}
+
+/// A helper to create guest attributes, which are normally passed as a
+/// `CFDictionary` with varying types.
+pub struct GuestAttributes {
+    inner: CFMutableDictionary,
+}
+
+impl GuestAttributes {
+    // Not implemented:
+    // - architecture
+    // - canonical
+    // - dynamic code
+    // - dynamic code info plist
+    // - hash
+    // - mach port
+    // - sub-architecture
+
+    /// The guest's audit token.
+    pub fn audit_token(&mut self, token: CFDataRef) {
```

Comment:
I think they should be named `set_*`. Other than that it seems fine. 

Maybe add `set_other<V: ToVoid<V>>(CFStringRef, V)` to allow adding other attributes.